### PR TITLE
* [tosa] Support for AtenReshapeOp

### PR DESCRIPTION
**Pull request by anupgangwar**
* [tosa] Support for AtenBatchNormOp

Signed-off-by: Anup Gangwar <anup.gangwar@arm.com>

### CODE REVIEW

**Comment by anupgangwar** [comment]
```
Summary:
    Passed: 62
    Expectedly Failed: 189
```
Tagging @cathyzhyi for review.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 1]
```diff
@@ -1760,6 +1760,166 @@ LogicalResult ConvertAtenOp<AtenConv2dOp>::matchAndRewrite(
...
+  return success();
+}
+
+// Normalization ops perform elementwise ops of a single mean/stdev value
+// against the feature map and because input is NCHW, the rank-1 value must be
+// reshaped so it sits on the same dim as 'C'.
+static LogicalResult reshapeToNormInputDim(Operation *op,
+                                           ConversionPatternRewriter &rewriter,
+                                           TypeConverter *converter,
+                                           Type outType, const Value &toBcast,
```
Comment (cathyzhyi) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:1801]: nit: don't need & when passing Value type. Value type itself is already a pointer wrapper of the storage type.

**Review by cathyzhyi — CHANGES_REQUESTED** [review]

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 2]
```diff
@@ -1760,6 +1760,166 @@ LogicalResult ConvertAtenOp<AtenConv2dOp>::matchAndRewrite(
...
+  // op5 = mul(op4, bscale)
+  // op6 = add(op5, boffset)
+
+  auto meanType = adaptor.running_mean().getType().dyn_cast<TensorType>();
+  auto varianceType = adaptor.running_var().getType().dyn_cast<TensorType>();
+  if (!varianceType || !meanType)
+    return op.emitError("Only ranked tensor types are supported");
+
+  Value meanVal, varianceVal, weightVal, biasVal;
+  if (meanType.getNumElements() == 0) {
```
Comment (cathyzhyi) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:1864]: I think this doesn't really exist right? Maybe we could just assert the getNumElements != 0?

Comment (anupgangwar) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:1864]: Would that be the case for varianceType as well?

Comment (cathyzhyi) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:1864]: Yea, I think so.

**Comment by anupgangwar** [comment]
@cathyzhyi All changes have been incorporated.

**Review by cathyzhyi — APPROVED** [review]

**Pull request merged**